src/taxes/views/ChannelsList.tsx

```diff
@@ -0,0 +1,37 @@
+import useNavigator from "@saleor/hooks/useNavigator";
+import useShop from "@saleor/hooks/useShop";
+import React from "react";
+
+import TaxChannelsPage from "../components/TaxChannelsPage";
+import { taxConfigurations } from "../fixtures";
+import { channelsListUrl, taxTabSectionUrl } from "../urls";
+
+interface ChannelsListProps {
+  id: string;
+}
+
+export const ChannelsList: React.FC<ChannelsListProps> = ({ id }) => {
+  const navigate = useNavigator();
+  const shop = useShop();
+
+  const handleTabChange = (tab: string) => {
+    navigate(taxTabSectionUrl(tab));
+  };
+
+  React.useEffect(() => {
+    if (id === "undefined" && taxConfigurations) {
+      navigate(channelsListUrl(taxConfigurations[0].channel.id));
+    }
+  }, [id, taxConfigurations]);
+
```

Comment:
```suggestion
if (id === "undefined" && taxConfigurations) {
  return null;
}

```
After this change we don't render the `TaxChannelsPage` if we don't have required data to display it (btw this would fail if we had strictNullChecks 🙃)

Reply:
I don't get it. `id === "undefined"` clearily means wrong URL and 404 should be displayed.

Reply:
Extracted this redirect logic to a custom hook. id === undefined is valid, because we don't always know the id when we go to taxes view - for example from configuration view.